.metadata/generate.proj

```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project Sdk="Microsoft.Windows.WinmdGenerator/0.39.18-preview">
+    <PropertyGroup Label="Globals">
+        <OutputWinmd>../.windows/winmd/Windows.Win32.Graphics.Direct3D12.winmd</OutputWinmd>
+        <WinmdVersion>1.608.0</WinmdVersion>
+        <IdlDir>../include/directx</IdlDir>
+    </PropertyGroup>
+    <ItemGroup>
+        <Idls Include="$(IdlDir)/dxgicommon.idl;$(IdlDir)/dxgiformat.idl;$(IdlDir)/d3dcommon.idl;$(IdlDir)/d3d12.idl;$(IdlDir)/d3d12compatibility.idl" />
+
+        <Partition Include="dxgicommon.cpp">
+            <TraverseFiles>$(CompiledHeadersDir)/dxgiformat.h;$(CompiledHeadersDir)/dxgicommon.h</TraverseFiles>
+            <ExcludeFromCrossarch>true</ExcludeFromCrossarch>
+            <Namespace>Windows.Win32.Graphics.Dxgi.Common</Namespace>
+        </Partition>
+
+        <Partition Include="d3dcommon.cpp">
+            <TraverseFiles>$(CompiledHeadersDir)/d3dcommon.h</TraverseFiles>
+            <ExcludeFromCrossarch>true</ExcludeFromCrossarch>
+            <Namespace>Windows.Win32.Graphics.Direct3D</Namespace>
+        </Partition>
+
+        <Partition Include="d3d12.cpp">
+            <TraverseFiles>$(CompiledHeadersDir)/d3d12.h;$(CompiledHeadersDir)/d3d12sdklayers.h;$(CompiledHeadersDir)/d3d12compatibility.h;$(IdlDir)/d3d12shader.h</TraverseFiles>
```

Comment:
I've added `d3d12compatibility` (based on IDL) and `d3d12shader` (based on the header file in the repo as there's no IDL (makes me wonder if I could/should generate everything from `.h` instead of IDL).

Unfortunately `d3d12sdklayers.h` doesn't want to take. I cannot generate that header from IDL as it is included halfway in `d3d12.idl` and depends on prior definitions, and I guess that makes the IDL compiler _not_ spit out an intermediary `$(CompiledHeadersDir)/d3d12sdklayers.h` (while compiling `$(IdlDir)/d3d12.idl`), hence causing this file-to-be-traversed to not exist?